### PR TITLE
feat: Add authentication DEV-2296

### PR DESCRIPTION
**Pull request by seakayone**
Make service endpoints protected. 
Needs and `Authorization` header with a jwt `Bearer` token in the request.

Integrate auth into main app with option to "disable" auth by setting env var `JWT_DISABLE_AUTH` to `true` and provide  an auth bearer token with an arbitrary value with the request.

Update to Scala 3.3.0

### CODE REVIEW

File: src/main/scala/swiss/dasch/api/Authenticator.scala [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  def authenticate(token: String): ZIO[Any, NonEmptyChunk[AuthenticationError], JwtClaim]
+}
+object Authenticator {
+
+  val middleware: RequestHandlerMiddleware[Nothing, Authenticator with JwtConfig, Nothing, Any] =
+    HttpAppMiddleware.bearerAuthZIO(token =>
+      for {
+        isAuthDisabled  <- ZIO.serviceWith[JwtConfig](_.disableAuth)
+        isAuthenticated <- if (isAuthDisabled) { ZIO.succeed(true) }
+                           else { Authenticator.authenticate(token).fold(_ => false, _ => true) }
```
Comment (mpro7) [inline src/main/scala/swiss/dasch/api/Authenticator.scala:25]:
```suggestion
        isAuthenticated <- if (isAuthDisabled) ZIO.succeed(true)
                           else Authenticator.authenticate(token).fold(_ => false, _ => true)
```

File: src/main/scala/swiss/dasch/api/Authenticator.scala [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+                 .refineOrDie { case e: JwtException => NonEmptyChunk(AuthenticationError.jwtProblem(e)) }
+      _     <- verifyClaim(claim)
+    } yield claim
+
+  private def verifyClaim(claim: JwtClaim): IO[NonEmptyChunk[AuthenticationError], JwtClaim] =
+    val audVal = if (claim.audience.getOrElse(Set.empty).contains(audience)) { Validation.succeed(claim) }
+    else { Validation.fail(AuthenticationError.invalidAudience(jwtConfig)) }
+
+    val issVal = if (claim.issuer.contains(issuer)) { Validation.succeed(claim) }
+    else { Validation.fail(AuthenticationError.invalidIssuer(jwtConfig)) }
```
Comment (mpro7) [inline src/main/scala/swiss/dasch/api/Authenticator.scala:64]: unnecessary if statement brackets

**Review by mpro7 — APPROVED** [review]
Some new Scala files don't have copyright header and feel free to apply below suggestions/ Thanks.

**Comment by seakayone** [comment]
> Some new Scala files don't have copyright header 

Thanks for noting. I have added `headerCheckAll` to our build. So it should fail next time we are missing the headers.

https://github.com/dasch-swiss/dsp-ingest/pull/15/commits/ac0469c9a2328abdb8b032ee75a0a4794c124494

**Pull request merged**